packages/starlight/integrations/expressive-code.ts

```diff
@@ -0,0 +1,265 @@
+	const { locales, defaultLocale, expressiveCode } = opts;
+	if (expressiveCode === false) return [];
+	const config = typeof expressiveCode === 'object' ? expressiveCode : {};
+
+	const {
+		theme = ['github-dark', 'github-light'],
+		customizeTheme,
+		useThemedSelectionColors = false,
+		styleOverrides: {
+			textMarkers: textMarkersStyleOverrides,
+			...otherStyleOverrides
+		} = {},
+		useStarlightDarkModeSwitch,
+		useStarlightUiThemeColors = config.theme === undefined,
+		minSyntaxHighlightingColorContrast = 5.5,
+		plugins = [],
+		...rest
+	} = config;
+
+	const configuredThemesCount = (Array.isArray(theme) && theme.length) || 1;
+	if (useStarlightDarkModeSwitch !== false) {
+		const loadedThemes: ExpressiveCodeTheme[] = [];
+		const darkModeSwitchCss: string[] = [];
+		plugins.push({
+			name: 'Starlight Dark Mode Switch',
+			baseStyles: ({ theme, themeClassName }) => {
+				loadedThemes.push(theme);
+				darkModeSwitchCss.push(`
+					:root:not([data-theme='${theme.type}']) .${themeClassName} {
+						display: none;
+					}
+				`);
+				if (loadedThemes.length !== configuredThemesCount) return '';
+				const foundDarkThemes = loadedThemes.filter((t) => t.type === 'dark').length;
+				const foundLightThemes = loadedThemes.filter((t) => t.type === 'light').length;
+				const configuredThemesSupportDarkModeSwitch =
+					foundDarkThemes === 1 && foundLightThemes === 1;
+				if (!configuredThemesSupportDarkModeSwitch) {
+					if (useStarlightDarkModeSwitch === true)
+						console.warn(
+							`*** Warning: To support the config option "useStarlightDarkModeSwitch: true", please provide 1 dark and 1 light theme (found ${foundDarkThemes} and ${foundLightThemes}).\n`
+						);
+					return '';
+				}
+				return darkModeSwitchCss.join('\n');
+			},
+		});
+	}
+
+	if (useStarlightUiThemeColors === true && configuredThemesCount < 2)
+		console.warn(`*** Warning: Using the config option "useStarlightUiThemeColors: true" with only one theme is not recommended. For better color contrast, please provide 1 dark and 1 light theme.\n`);
+
+	// Add Expressive Code UI translations (if any) for all defined locales
+	if (locales) {
+		for (const locale in locales) {
+			const lang = locales[locale]?.lang;
+			if (!lang) continue;
+
+			addTranslations(lang, pluginFramesTexts, [
+				'expressiveCode.copyButtonCopied',
+				'expressiveCode.copyButtonTooltip',
+				'expressiveCode.terminalWindowFallbackTitle',
+			]);
+		}
+	}
+
+	return [
+		astroExpressiveCode({
+			theme,
+			customizeTheme: (theme) => {
+				if (useStarlightUiThemeColors) {
+					applyStarlightUiThemeColors(theme);
+				}
+				if (customizeTheme) {
+					theme = customizeTheme(theme) ?? theme;
+				}
+				if (minSyntaxHighlightingColorContrast > 0) {
+					ensureMinTokenColorContrast(theme, minSyntaxHighlightingColorContrast);
+				}
+				return theme;
+			},
+			useThemedSelectionColors,
+			styleOverrides: {
+				codeFontFamily: 'var(--__sl-font-mono)',
```

Comment:
I believe APIs might change slightly but documenting here some tweaks I made to align the design closer to our Figma mock-ups:

```suggestion
				borderRadius: '0',
				borderWidth: '1px',
				codePaddingBlock: '0.75rem',
				codePaddingInline: '1rem',
				codeFontFamily: 'var(--__sl-font-mono)',
```

Reply:
I'm not sure if I speak for everybody - I now opinions are different -, but personally I liked the rounded corners a lot more actually... I used them on my website already and the look fantastic, especially sh and bash with the terminal look. But I guess if I want rounded corners, I have to edit the settings locally...

Reply:
Yes, this change would just align the default design more with all our other defaults but you could still set options to configure this to your taste and customise how things look.

Reply:
I added the proposed styles. However, I'd like to note that your reduced inline padding does not leave much space for the `+`/`-` diff indicators anymore. I think it's still okay in this state, but any less and it wouldn't fit anymore. :)